fuzzing/nxt_http_h1p_fuzz.c

```diff
@@ -50,32 +50,58 @@ LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
         return 0;
     }
 
+    buf.start = (u_char *) data;
+    buf.end = (u_char *) data + size;
+    buf.pos = buf.start;
+    buf.free = buf.end;
+
+
+    /*
+     * Create memory pool.
+     * Initialize 'nxt_http_request_t'
+     */
     mp = nxt_mp_create(1024, 128, 256, 32);
     if (mp == NULL) {
         return 0;
     }
 
-    nxt_memzero(&rp, sizeof(nxt_http_request_parse_t));
-    if (nxt_http_parse_request_init(&rp, mp) != NXT_OK) {
+    r_h1p = nxt_mp_zget(mp, sizeof(nxt_http_request_t));
+    if (r_h1p == NULL) {
         goto failed;
     }
 
-    buf.start = (u_char *)data;
-    buf.end = (u_char *)data + size;
-    buf.pos = buf.start;
-    buf.free = buf.end;
-
-    if (nxt_http_parse_request(&rp, &buf) != NXT_DONE) {
+    r_h1p->proto.h1 = nxt_mp_zget(mp, sizeof(nxt_h1proto_t));
+    if (r_h1p->proto.h1 == NULL) {
         goto failed;
     }
 
-    r_h1p = nxt_mp_zget(mp, sizeof(nxt_http_request_t));
+    r_h1p->conf = nxt_mp_zget(mp, sizeof(nxt_socket_conf_joint_t));
+    if (r_h1p->conf == NULL) {
+        goto failed;
+    }
 
-    if (r_h1p == NULL) {
+    r_h1p->conf->socket_conf = nxt_mp_zget(mp, sizeof(nxt_socket_conf_t));
+    if (r_h1p->conf->socket_conf == NULL) {
         goto failed;
     }
 
     r_h1p->mem_pool = mp;
+    r_h1p->conf->socket_conf->max_body_size = 8 * 1024 * 1024;
+
+
+    /*
+     * Process request and fields.
+     * Parse http cookies and arguments. 
+     */
+    nxt_memzero(&rp, sizeof(nxt_http_request_parse_t));
+
+    if (nxt_http_parse_request_init(&rp, mp) != NXT_OK) {
+        goto failed;
+    }
+
+    if (nxt_http_parse_request(&rp, &buf) != NXT_DONE) {
```

Comment:
I would generally prefer to do the assignment and then the check in separate steps.

Reply:
Ah... `NXT_DONE` does need an extra step. Thanks.